code/modules/admin/verbs/selectequipment.dm

```diff
@@ -0,0 +1,230 @@
+	var/icon/dummysprite = get_flat_human_icon(null, prefs=prefs, dummy_key = dummy_key, outfit_override = selected_outfit)
+	data["icon64"] = icon2base64(dummysprite)
+	data["name"] = target_mob
+
+	data["favorites"] = list()
+	if(prefs)
+		data["favorites"] = prefs.favorite_outfits
+
+	var/list/custom
+	custom += make_custom_outfit_entries(GLOB.custom_outfits)
+	data["custom_outfits"] = custom
+	data["current_outfit"] = selected_identifier
+	return data
+
+
+/datum/select_equipment/ui_static_data(mob/user)
+	var/list/data = list()
+	if(!cached_outfits)
+		cached_outfits = list()
+		cached_outfits += list(outfit_entry("General", /datum/outfit, "Naked", TRUE))
+		cached_outfits += make_outfit_entries("General", subtypesof(/datum/outfit) - typesof(/datum/outfit/job) - typesof(/datum/outfit/plasmaman))
+		cached_outfits += make_outfit_entries("Jobs", typesof(/datum/outfit/job))
+		cached_outfits += make_outfit_entries("Plasmamen Outfits", typesof(/datum/outfit/plasmaman))
+
+	data["outfits"] = cached_outfits
+	return data
+
+
+/datum/select_equipment/proc/resolve_outfit(text)
+
+	var/path = text2path(text)
+	if(ispath(path, /datum/outfit))
+		return path
+
+	else //don't bail yet - could be a custom outfit
+		var/datum/outfit/custom_outfit = locate(text)
+		if(istype(custom_outfit))
+			return custom_outfit
+
+
+/datum/select_equipment/ui_act(action, params)
+	if(..())
+		return
+	. = TRUE
+	switch(action)
+		if("preview")
+			var/datum/outfit/new_outfit = resolve_outfit(params["path"])
+
+			if(ispath(new_outfit)) //got a typepath - that means we're dealing with a normal outfit
+				selected_identifier = new_outfit //these are keyed by type
+				//by the way, no, they can't be keyed by name because many of them have duplicate names
+
+			else if(istype(new_outfit)) //got an initialized object - means it's a custom outfit
+				selected_identifier = REF(new_outfit) //and the outfit will be keyed by its ref (cause its type will always be /datum/outfit)
+
+			else //we got nothing and should bail
+				return
+
+			selected_outfit = new_outfit
+
+		if("applyoutfit")
+			var/datum/outfit/new_outfit = resolve_outfit(params["path"])
+			if(new_outfit && ispath(new_outfit)) //initialize it
+				new_outfit = new new_outfit
+			if(!istype(new_outfit))
+				return
+			user.admin_apply_outfit(target_mob, new_outfit)
+
+		if("customoutfit")
+			user.outfit_manager()
+
+		if("togglefavorite")
+			var/datum/outfit/outfit_path = resolve_outfit(params["path"])
+			if(!ispath(outfit_path)) //we do *not* want custom outfits (i.e objects) here, they're not even persistent
+				return
+
+			if(user.prefs.favorite_outfits.Find(outfit_path)) //already there, remove it
+				user.prefs.favorite_outfits -= outfit_path
+			else //not there, add it
+				user.prefs.favorite_outfits += outfit_path
+			user.prefs.save_preferences()
+
+/client/proc/admin_apply_outfit(mob/target, dresscode)
+	if(!(ishuman(target) || isobserver(target)))
```

Comment:
```suggestion
	if(!ishuman(target) || !isobserver(target))
```

Reply:
Then that check will never work

Reply:
This is exactly the same as the check you have now.

Reply:
Oh wait, my mistake. I updated the suggestion.